### PR TITLE
Handle Push Notification

### PR DESCRIPTION
**Pull request by gguijarro-c-chwy**
This code handles push notification. 

**Requirements**

'incoming_transaction':true 

Must be passed in the push notification data. 
Only handles foreground push notifications

The ESR link must be passed in the notification message 

image
image

### CODE REVIEW

**Comment by gguijarro-c-chwy** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>:exclamation: PR has more than 500 line changes </td>
    </tr>
  </tbody>
</table>



> Pull Request size seems relatively large. If Pull Request contains multiple changes, split each into separate PR will helps faster, easier review.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a6ceeef8b8c788825c4b3dea039476082a4be356
</p>

File: lib/ui/blocs/deeplink/deeplink_bloc.dart [hunk 1]
```diff
@@ -11,21 +11,13 @@ part 'deeplink_state.dart';
 part 'page_command.dart';
 
 class DeeplinkBloc extends Bloc<DeeplinkEvent, DeeplinkState> {
```
Comment (n13) [inline lib/ui/blocs/deeplink/deeplink_bloc.dart:13]: Does this disable deep links?

How are deep links handled now?

File: lib/ui/settings/interactor/settings_bloc.freezed.dart [hunk 2]
```diff
@@ -1417,7 +1417,7 @@ class __$$_SettingsStateCopyWithImpl<$Res>
 
 class _$_SettingsState implements _SettingsState {
   const _$_SettingsState(
```
Comment (n13) [inline lib/ui/settings/interactor/settings_bloc.freezed.dart:1419]: part of this PR?

File: pubspec.lock [hunk 3]
```diff
@@ -13,10 +13,10 @@ packages:
     dependency: transitive
     description:
       name: _flutterfire_internals
-      sha256: cb3a948a1eebbf8efd987c43f95418269930e912a88bc7b6a5a7658423133635
```
Comment (n13) [inline pubspec.lock:16]: why do we need sha hash version?

**Review by n13 — APPROVED** [review]
Approved with comments

File: lib/ui/blocs/deeplink/deeplink_bloc.dart (hunk 1 above)
Comment (gguijarro-c-chwy) [reply lib/ui/blocs/deeplink/deeplink_bloc.dart:13]: this code was never used

File: lib/ui/settings/interactor/settings_bloc.freezed.dart (hunk 2 above)
Comment (gguijarro-c-chwy) [reply lib/ui/settings/interactor/settings_bloc.freezed.dart:1419]: No, This was generated when i called `flutter pub run build_runner` looks like we committed code to main without running the `flutter pub run build_runner`

**Comment by n13** [comment]
Ok I see - no problem

**Pull request merged**